File: sep-product-service/src/main/java/com/kurtcan/sepproductservice/product/Product.java
package com.kurtcan.sepproductservice.product;

import com.kurtcan.sepproductservice.shared.entity.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import lombok.*;
import lombok.experimental.SuperBuilder;

import java.math.BigDecimal;

@Data
@Entity
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
@ToString(callSuper = true)
@EqualsAndHashCode(callSuper = true)
public class Product extends BaseEntity {
    @Column(name = "name", length = 250, nullable = false)
    private String name;

    @Column(name = "description", length = 1000, nullable = false)
    private String description;

    @Column(name = "price", nullable = false)
    private BigDecimal price;

    @Column(name = "stock_amount", nullable = false)
    private int stockAmount;
}

File: sep-search-service/src/main/kotlin/com/kurtcan/sepsearchservice/product/Product.kt
package com.kurtcan.sepsearchservice.product

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.kurtcan.sepsearchservice.base.BaseEntity
import java.math.BigDecimal

data class Product @JsonCreator constructor(
    @JsonProperty("name") val name: String,
    @JsonProperty("description") val description: String,
    @JsonProperty("price") val price: BigDecimal,
    @JsonProperty("stockAmount") var stockAmount: Int
) : BaseEntity()